omnigibson/systems/macro_particle_system.py

```diff
@@ -801,10 +793,10 @@ def _sync_particle_groups(cls, group_objects, particle_idns, particle_attached_l
             "link_names": link_names,
         }
             for obj, p_idns, link_names in
-            zip(group_objects, particle_idns, particle_attached_link_names) if obj is not None}
+            zip(group_objects, particle_idns, particle_attached_link_names)}
```

Comment:
did we verify this works under our tests / examples?

Reply:
especially sim_save_load demo script

Reply:
done